Source/AwsCommonRuntimeKit/auth/signing/SigV4HttpRequestSigner.swift

```diff
@@ -53,53 +51,43 @@ public class SigV4HttpRequestSigner {
-        }
-        let configPtr = UnsafePointer(base)
-
-        let callbackPointer: UnsafeMutablePointer<SigningCallbackData> = fromPointer(ptr: callbackData)
-
-        defer {
-            base.deinitializeAndDeallocate()
-        }
-
-        aws_sign_request_aws(allocator.rawValue,
-                             signable,
-                             configPtr, { (signingResult, errorCode, userData) -> Void in
-            guard let userData = userData else {
-                return
-            }
-            let callback = userData.assumingMemoryBound(to: SigningCallbackData.self)
-            defer {
-                aws_signable_destroy(callback.pointee.signable)
-                callback.deinitializeAndDeallocate()
-            }
-
-            if let continuation = callback.pointee.continuation {
-                if errorCode == 0,
-                   let signingResult = signingResult {
-
-                    let signedRequest = aws_apply_signing_result_to_http_request(callback.pointee.request.rawValue,
-                                                                                 callback.pointee.allocator.rawValue,
-                                                                                 signingResult)
-                    if signedRequest == 0 {
-                        continuation.resume(returning: callback.pointee.request)
-                    } else {
-                        continuation.resume(throwing: CommonRunTimeError
-                                .crtError(CRTError(code: signedRequest)))
+        //TODO: fix callback
+        config.withCPointer { configPointer in
+            configPointer.withMemoryRebound(to: aws_signing_config_base.self, capacity: 1) { configBasePointer in
+                let callbackPointer: UnsafeMutablePointer<SigningCallbackData> = fromPointer(ptr: callbackData)
+                aws_sign_request_aws(allocator.rawValue,
+                        signable,
+                        configBasePointer, { (signingResult, errorCode, userData) -> Void in
+                    guard let userData = userData else {
+                        return
+                    }
+                    let callback = userData.assumingMemoryBound(to: SigningCallbackData.self)
+                    defer {
+                        aws_signable_destroy(callback.pointee.signable)
+                        callback.deinitializeAndDeallocate()
                     }
-                } else {
-                    continuation.resume(throwing: CommonRunTimeError
-                            .crtError(CRTError(code: errorCode)))
-                }
-            }
 
-        }, callbackPointer)
+                    if let continuation = callback.pointee.continuation {
+                        if errorCode == 0,
+                           let signingResult = signingResult {
+
+                            let signedRequest = aws_apply_signing_result_to_http_request(callback.pointee.request.rawValue,
```

Comment:
sorry, commenting on old code... feel free to TODO this in the future

I haven't looked into what circumstance `continuation` might not be set ...
but don't we ALWAYS want to do `aws_apply_signing_result_to_http_request()`, even if continuation isn't set?

Reply:
Continuation should always be set as we pass it as part of user data. Will refactor when I refactor this class to use core classes and learning more about it.

Reply:
This is a side effect of treating user data as nullable which will be fixed by !

Reply:
I have refactored it a bit to remove some confusion. I will refactor it properly to use core class etc in the Auth Layer refactor PR.